plonky2/src/iop/challenger.rs

```diff
@@ -146,6 +146,10 @@ impl<F: RichField, H: Hasher<F>> Challenger<F, H> {
         self.output_buffer
             .extend_from_slice(&self.sponge_state[0..SPONGE_RATE]);
     }
+
+    pub fn state(&self) -> [F; SPONGE_WIDTH] {
```

Comment:
Maybe call `duplexing` in here? Seems like we'd never want to get the state without first processing the input buffer.